app/src/main/java/com/hedvig/app/feature/offer/OfferActivity.kt

```diff
@@ -69,7 +79,7 @@ class OfferActivity : BaseActivity(R.layout.activity_offer) {
                     })
                 } else {
                     val completeQuote = data.lastQuoteOfMember.asCompleteQuote!!
-                    offerViewModel.fetchPreSale(completeQuote.toContractType())
+                    perilBinder.contract = completeQuote.toContractType()
```

Comment:
Let's provide the `PerilBinder` with the contract type at the point of `bind()`. The `bind()`-method can even take the larger `OfferQuery.Data`-object as it binds on multiple points of its data 🙂